sdk/schemaregistry/schema-registry-json/review/schema-registry-json.api.md

```diff
@@ -6,10 +6,18 @@
 
 import { SchemaRegistry } from '@azure/schema-registry';
 
+// @public
+export interface DeserializeOptions {
+    validateCallback?: (message: string, schema: string) => boolean | {
+        isValid: false;
+        message: string;
```

Comment:
Is the name not clear enough? should it be `error` or `errorMessage`? Also did we consider returning the error object rather than just the error message? /cc @bterlson

Reply:
I like `errorMessage`.

Had not deeply considered returning the error object. I'm not necessarily opposed, though I have a slight worry that treating any object-like return as an error could be more lossy. I believe replacing `return` with `throw` in that case will just do what you want, which was an original design we considered, but for reasons I've forgotten it was felt that an ability to throw an error rather than return one would be nice. I wouldn't be opposed to removing the object return for now and waiting for feedback I suppose.

Reply:
IIRC the reason we move to `return` instead of `throw` is that the customers might get confused with `void` function signature, so we want to have `errorMessage` or `errorObject` showing up in our function signature.

Reply:
That makes sense, thanks!

@minhanh-phan I will leave the object return vs throw decision to you but if you go with the former, let's do the `errorMessage` rename.